examples/stabilizer_code.py

```diff
@@ -0,0 +1,192 @@
+import numpy as np
+
+import cirq
+
+# Based on:
+# Stabilizer Codes and Quantum Error Correction
+# Daniel Gottesman
+# https://thesis.library.caltech.edu/2900/2/THESIS.pdf
+
+
+class StabilitizerCode(object):
+    def __init__(self, group_generators):
+        n = len(group_generators[0])
+        k = n - len(group_generators)
+
+        # Build the matrix defined in section 3.4
+        M = np.zeros((n - k, 2 * n), np.int8)
+        for i, group_generator in enumerate(group_generators):
+            for j, c in enumerate(group_generator):
+                if c == 'X' or c == 'Y':
+                    M[i, j] = 1
+                elif c == 'Z' or c == 'Y':
+                    M[i, n + j] = 1
+
+        # Performing the Gaussian elimination as in section 4.1
```

Comment:
I had looked for it, but it seemed difficult because the Gaussian elimination must be done in GF(2) not in R. Another difference is that the pivot must be chosen only in the sub-matrix E of equation 4.1.

Leaving unresolved for now.

Reply:
That seems fair to me. It might be worthwhile to note this in a docstring for `_GaussianElimination` above.

Reply:
Done.